app/routers/nutrition_log.py

```
@@ -61,42 +88,114 @@ def _event_from_day_close(payload: DayCloseRequest) -> DomainEvent:
+    except IntegrityError as e:
+        session.rollback()
+        # Narrowed idempotency check: only treat as replay if it's the idempotency constraint
+        if _is_idempotency_violation(e):
+            # Day already closed - skip re-finalization, return current state
+            already_closed = True
+        else:
+            # Other IntegrityError (FK, check, etc.) - propagate
+            raise
+
+    # Collect day events for finalization context
+    collected = collect_day_events(session, subject_id, day)
```

Flake8 (7.3.0)

[error] local variable 'collected' is assigned to but never used

(F841)